### PR TITLE
feat(venv): support pip options

### PR DESCRIPTION
**Pull request by rickzx**
- Support pip options when creating venv. Now `requirements.txt` can have options like
```
--extra-index-url https://flashinfer.ai/whl/cu121/torch2.3
...
```

- Support setting env vars if specified in bento.yaml. This is needed for, e.g., running gemma-2 model with
```
VLLM_ATTENTION_BACKEND=FLASHINFER
```

- Reformat all Python files with `black`

### CODE REVIEW

File: src/openllm/local.py [hunk 1]
```diff
@@ -16,12 +16,25 @@ def _get_serve_cmd(bento: BentoInfo, port: int = 3000):
...
 def serve(bento: BentoInfo, port: int = 3000):
+    prep_env_vars(bento)
     venv = ensure_venv(bento)
     cmd, env, cwd = _get_serve_cmd(bento, port=port)
     output(f'Access the Chat UI at http://localhost:{port}/chat (or with you IP)')
     run_command(cmd, env=env, cwd=cwd, venv=venv)
 
 
+def prep_env_vars(bento: BentoInfo):
+    import os
```
Comment (aarnphm) [inline src/openllm/local.py:27]: we should move this to top of files.

**Review by aarnphm — COMMENTED** [review]
shouldn't ruff format be sufficient enough?

**Comment by rickzx** [comment]
> shouldn't ruff format be sufficient enough?

Yeah, i realized the CI is already doing this. No need for black format

**Review by aarnphm — APPROVED** [review]

File: src/openllm/local.py [hunk 2]
```diff
@@ -16,6 +28,7 @@ def _get_serve_cmd(bento: BentoInfo, port: int = 3000):
 
 
 def serve(bento: BentoInfo, port: int = 3000):
+    prep_env_vars(bento)
     venv = ensure_venv(bento)
     cmd, env, cwd = _get_serve_cmd(bento, port=port)
```
Comment (bojiang) [inline src/openllm/local.py:33]: It seems the prepared env isn't applied to `bentoml serve`? @rickzx

Comment (rickzx) [reply src/openllm/local.py:33]: It is applied to `bentoml serve`. The `run_command` function explicitly copies the env vars from the current process to the new process (even if it doesn't, the default subprocess will also inherit env from the current process)

**Pull request merged**